guide/sharding/additional-information.md

```diff
@@ -75,50 +77,50 @@ You can access them later as usual via `process.argv`, which contains an array o
 
 ## Eval arguments
 
-There may come the point where you will want to pass functions or arguments from the outer scope into a `.broadcastEval()` call.
+There may come the point where you will want to pass arguments from the outer scope into a `.broadcastEval()` call.
 
 ```js
-client.shard.broadcastEval(`(${funcName})('${arg}')`);
-```
-
-In this small snippet, an entire function is passed to the eval. It needs to be encased in parenthesis; it will throw errors on its way there otherwise. Another set of parenthesis is required so that the function gets called. Finally, the passing of the argument itself, which slightly varies, depending on the type of argument you are passing. If it's a string, you must wrap it in quotes, or it will be interpreted as is and will throw a syntax error because it won't be a string by the time it gets there.
+function funcName(client, { arg }) {
+	// ...
+}
 
-Now, what if you wanted to call a function from *within* the client context? That is to say, you are not passing a function. It would look something like this:
-
-```js
-client.shard.broadcastEval(`this.${funcName}(${args});`);
+client.shard.broadcastEval(funcName, { context: { arg: 'arg' } });
 ```
 
-This would become `client.funcName(args)` once it gets through. This is handy if you, for example, have extended your client object with your class and wish to call some of its methods manually.
+In this small snippet, an argument is passed to the `funcName` function through the `context` option of the <DocsLink path="typedef/BroadcastEvalOptions">`BroadcastEvalOptions`</DocsLink>.
+The function will recieve the arguments as an object as the second parameter.
+
+::: warning
+The `context` option only accepts properties which are JSON-serializable.
+This means you can not pass complex data types in the context directly.
+:::
 
 ### Asynchronous functions
 
 There may be a time when you want to have your shard process an asynchronous function. Here's how you can do that!
 
 ```js
-client.shard.broadcastEval(`
-	let channel = this.channels.cache.get('id');
+client.shard.broadcastEval(client => {
+	const channel = this.channels.cache.get('id');
```

Comment:
I have no idea what this code it, but for the sake of everyone's sanity:

```javascript
const message = await client.channels.cache.get('id')?.messages.fetch('id');
return message?.id;
```

Reply:
This portion probably needs merging and rewriting, since async functions are now straight-forward, unlike v12's broadcastEval.

Reply:
Actually, I think this section should be removed, then.

Reply:
Agreed, the entire "Asynchronous functions" section should be removed

Reply:
Removed 👍